### DISCUSSION TITLE
can we training emformer transducer stateless with phoneme base?

### DISCUSSION THREAD

**Discussion by trangtv57** [k2-fsa/icefall#585] [Q&A] [answered]
I can see all script of you training transducer using sentence piece model. But As traditional model like kaldi using tdnn-lstm, I have trained model using phoneme, It's can make my system can decode with LG later, It's help me can control output using ngram+lexicon like old kaldi. But i'm not sure it's good or not. can give me some advice

**Comment by yfyeung** [comment, marked as answer]
Actually, icefall supports model training with phoneme lexicon.
Take LibriSpeech for example:
To prepare data:  https://github.com/k2-fsa/icefall/blob/master/egs/librispeech/ASR/prepare.sh#L161-L189
To generate a unique lexicon: https://github.com/k2-fsa/icefall/blob/master/egs/librispeech/ASR/local/generate_unique_lexicon.py
To use unique lexicon duration training or decoding: https://github.com/k2-fsa/icefall/blob/master/icefall/lexicon.py#L200
